it/controllers/pensions/WorkplaceAmountControllerISpec.scala

```diff
@@ -0,0 +1,430 @@
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY), captionSelector)
+          textOnPageCheck(expectedParagraph, paragraphSelector(2))
+          textOnPageCheck(user.specificExpectedResults.get.expectedBullet1, bulletListSelector(1))
+          textOnPageCheck(user.specificExpectedResults.get.expectedBullet2, bulletListSelector(2))
+          textOnPageCheck(user.specificExpectedResults.get.expectedYouCanFindThisOut, paragraphSelector(4))
+          textOnPageCheck(hintText, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck(amountInputName, inputSelector, "")
+          buttonCheck(buttonText, continueButtonSelector)
+          formPostLinkCheck(workplacePensionAmount(taxYearEOY), formSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        "render how much did you pay into your workplace pensions amount page when cya data" which {
+
+          val existingAmount: String = "999.88"
+          lazy val result: WSResponse = {
+            dropPensionsDB()
+            authoriseAgentOrIndividual(user.isAgent)
+            val pensionsViewModel = aPaymentsIntoPensionViewModel.copy(
+              workplacePensionPaymentsQuestion = Some(true), totalWorkplacePensionPayments = Some(BigDecimal(existingAmount)))
+            insertCyaData(pensionsUsersData(isPrior = false, aPensionsCYAModel.copy(paymentsIntoPension = pensionsViewModel)), aUserRequest)
+            urlGet(fullUrl(workplacePensionAmount(taxYearEOY)), user.isWelsh, follow = false,
+              headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+          }
+
+          implicit def document: () => Document = () => Jsoup.parse(result.body)
+
+          "has an OK status" in {
+            result.status shouldBe OK
+          }
+          titleCheck(user.specificExpectedResults.get.expectedTitle)
+          h1Check(user.specificExpectedResults.get.expectedHeading)
+          captionCheck(expectedCaption(taxYearEOY), captionSelector)
+          textOnPageCheck(expectedParagraph, paragraphSelector(2))
+          textOnPageCheck(user.specificExpectedResults.get.expectedBullet1, bulletListSelector(1))
+          textOnPageCheck(user.specificExpectedResults.get.expectedBullet2, bulletListSelector(2))
+          textOnPageCheck(user.specificExpectedResults.get.expectedYouCanFindThisOut, paragraphSelector(4))
+          textOnPageCheck(hintText, hintTextSelector)
+          textOnPageCheck(poundPrefixText, poundPrefixSelector)
+          inputFieldValueCheck(amountInputName, inputSelector, existingAmount)
+          buttonCheck(buttonText, continueButtonSelector)
+          formPostLinkCheck(workplacePensionAmount(taxYearEOY), formSelector)
+          welshToggleCheck(user.isWelsh)
+        }
+
+        //TODO redirect to question page
+        "redirect to the PensionsSummary page if the question has not been answered" which {
```

Comment:
Can move the 3 redirect tests out of the foreach loop as its not testing any different content